crates/exex/src/event.rs

```diff
@@ -0,0 +1,13 @@
+use reth_primitives::BlockNumber;
+
+/// Events emitted by an ExEx.
+#[derive(Debug)]
+pub enum ExExEvent {
+    /// Highest block processed by the ExEx.
+    ///
+    /// The ExEx must guarantee that it will not require all earlier blocks in the future, meaning
+    /// that Reth is allowed to prune them.
+    ///
+    /// On reorgs, it's possible for the height to go down.
```

Comment:
Now that I'm thinking about it, is that actually possible? So that fork chain has a lower block height than canonical chain.

Reply:
in the sense that the exex might currently have finished e.g. block #100, we reorg, and the ExEx emits a new `FinishedHeight` even for every block it processes in the chain segment (so it might emit e.g. 95, 96, 97, 98, 99, 100 after it has already processed 100 once before)